Investigate CI error

See https://github.com/octokit/graphql-action/runs/691058363#step:8:23

Happy to accept a pull request to fix the problem, I don't have the time to investigate myself right now

great response from GitHub support

> Thank you for reporting this and sharing this case with us.
> 
> We inspected the specific log reference you shared:
> 
> ```
> ##[error]bad indentation of a mapping entry at line 2, column 19:
>       repository(owner: "octokit", name: "graphql-acti ...
> 
> ```
> 
> Although the workflow file defines [this step here](https://github.com/octokit/graphql-action/actions/runs/104137758/workflow#L35), this particular error isn't something specific to GitHub Actions --- it's actually an error thrown by a third-party dependency used by the specified Action. We researched this together with our Actions engineers and wanted to share our findings here.
> 
> We observed that this workflow run is associated with `a745445` which is tagged at `v2.2.0`. This tag is associated with a release that references <https://github.com/octokit/graphql-action/pull/17>. We reviewed the files changed in that pull request and see that the JavaScript-based action's `index.js` introduced `js-yaml`, a dependency that was not previously leveraged before:
> 
> <https://github.com/octokit/graphql-action/pull/17/files#diff-168726dbe96b3ce427e7fedce31bb0bcR2>
> 
> The `package.json` file was updated to include `js-yaml` at `^3.13.1` as a part of `devDependencies`:
> 
> <https://github.com/nodeca/js-yaml/tree/3.13.1>
> 
> We checked out the repository at that tag and searched for the error shown in the log reference:
> 
> ```
> # clone the repository & navigate into it
> git clone git@github.com:nodeca/js-yaml.git
> cd js-yaml
> 
> # check the repository at this tag
> git checkout 3.13.1
> 
> # use `grep` search for this error
> grep -rn "bad indentation of a mapping entry" lib
> lib/js-yaml/loader.js:1107:      throwError(state, 'bad indentation of a mapping entry');
> 
> ```
> 
> This error is wrapped in the following conditional block:
> 
> <https://github.com/nodeca/js-yaml/blob/665aadda42349dcae869f12040d9b10ef18d12da/lib/js-yaml/loader.js#L1106-L1111>
> 
> We haven't made any changes to the `actions/runner` or deployed anything that would signal this is a parser error on our end. However, I hope that this particular research may help you investigate what might be happening as far as parsing that query goes.

--- a/index.js
+++ b/index.js
@@ -32,9 +32,24 @@ function getAllInputs() {
   return Object.entries(process.env).reduce((result, [key, value]) => {
     if (!/^INPUT_/.test(key)) return result;
 
-    const inputName = key.toLowerCase() === "input_mediatype" ? "mediaType" : key.substr("INPUT_".length).toLowerCase();
+    const inputName =
+      key.toLowerCase() === "input_mediatype"
+        ? "mediaType"
+        : key.substr("INPUT_".length).toLowerCase();
+
+    // The js-yaml parser cannot handle the syntax of a multi-line GraphQL query very well,
+    // so we just leave it as-is.
+    // https://github.com/octokit/graphql-action/issues/21
+    if (inputName === `query`) {
+      result.query = value;
+      return result;
+    }
+
     result[inputName] = yaml.safeLoad(value);
-    result[inputName] = result[inputName] == parseInt(result[inputName], 10) ? parseInt(result[inputName], 10) : result[inputName];
+    result[inputName] =
+      result[inputName] == parseInt(result[inputName], 10)
+        ? parseInt(result[inputName], 10)
+        : result[inputName];
     return result;
   }, {});
 }

